remote-wallet/src/remote_wallet.rs

```diff
@@ -104,126 +55,239 @@ impl RemoteWalletManager {
+        &self,
+        manufacturer: &Manufacturer,
+    ) -> Vec<RemoteWalletInfo> {
+        self.devices
+            .read()
+            .iter()
+            .filter(|device| &device.info.manufacturer == manufacturer)
+            .map(|device| device.info.clone())
+            .collect()
+    }
+
+    /// Get the first available wallet of any type
+    ///
+    /// This is a convenience method that returns the first connected wallet,
+    /// regardless of its type. Useful when you just need any available wallet.
+    pub fn get_first_available_wallet(&self) -> Option<RemoteWalletInfo> {
+        self.devices
+            .read()
+            .first()
+            .map(|device| device.info.clone())
+    }
+
+    /// Attempt to connect to hardware wallets with polling within a time limit
+    ///
+    /// This method will continuously attempt to discover and connect to hardware wallets
+    /// until either devices are found or the maximum polling duration is exceeded.
+    ///
+    /// Returns `true` if at least one device was successfully connected, `false` otherwise
     pub fn try_connect_polling(&self, max_polling_duration: &Duration) -> bool {
         let start_time = Instant::now();
         while start_time.elapsed() <= *max_polling_duration {
-            if let Ok(num_devices) = self.update_devices() {
-                let plural = if num_devices == 1 { "" } else { "s" };
-                trace!("{num_devices} Remote Wallet{plural} found");
-                return true;
+            match self.update_devices() {
+                Ok(_new_device_count) => {
+                    let current_total = self.devices.read().len();
+                    if current_total > 0 {
+                        let plural = if current_total == 1 {
+                            LOG_DEVICE_SINGULAR
+                        } else {
+                            LOG_DEVICE_PLURAL
+                        };
+                        trace!("{current_total} Remote Wallet{plural} found");
+                        return true;
+                    }
+                }
+                Err(err) => {
+                    debug!("Error during device discovery: {err:?}");
+                    // Continue trying despite errors
+                }
             }
+
+            // Small delay to avoid excessive polling
+            std::thread::sleep(Duration::from_millis(100));
         }
+
+        debug!(
+            "Polling timeout reached. No devices found after {max_polling_duration:?}"
+        );
         false
     }
 }
 
-/// `RemoteWallet` trait
+/// Trait for hardware wallet implementations
+///
+/// This trait defines the interface that all hardware wallet implementations must provide.
+/// It includes methods for device initialization, public key derivation, and message signing.
+///
+/// # Type Parameters
+/// * `T` - The device info type (typically `hidapi::DeviceInfo`)
 #[allow(unused_variables)]
 pub trait RemoteWallet<T> {
```

Comment:
Feel free to move this trait into another file if you prefer, maybe `src/trait.rs`?